Linux build

Allcharles opened the pull request:
Added ability for `npm run make` command to build both linux (.deb) and windows (.exe) files and installers. Program does not yet have icons, but they have been added to a res folder. This also contains the basics of a Travis CI file for automated testing and compilation.
Fixes #17 
Fixes #13 (uses `npm run make` to generate programs, this is the default name for electron but can be changed later)

Allcharles committed:
Added space to application name

Allcharles committed:
Added icons

Allcharles committed:
Builds deb, Failes to run

Returns failed to locate startable electron application.

Allcharles committed:
Fixed npm start failure

Allcharles committed:
Working linux build

Allcharles committed:
Delete Logs\log4net-debug.log

Allcharles committed:
Updated linux_dependencies script

Moved root level scripts into a seperate file to distinguish them.

Allcharles committed:
Working Linux AP Desktop name

Allcharles committed:
Removed Broken Icons

Allcharles committed:
Initial TravisCI Script

Allcharles committed:
Removal of log4net-debug.log

Allcharles committed:
Test fix .travis.yml

atruskie reviewed (changes requested):
The Travis build was failing, but I see you've fixed it in the meantime 👍

Allcharles committed:
Added mono-complete to travis ci

Allcharles committed:
mono-complete added to packages

Allcharles committed:
Potential OS Fix

Allcharles committed:
Test fix xvfb

Allcharles committed:
Updated Version Number

This is mainly to update travis ci

Allcharles committed:
Added fakeroot library

Allcharles committed:
Fixed travic ci order

Updated app icon.

Allcharles committed:
Fix tests

Allcharles committed:
PR Review Changes

Allcharles committed:
Remove unneeded install

Allcharles committed:
Merge pull request #28 from QutEcoacoustics/travis-ci-setup

Travis ci setup

Allcharles committed:
Updated test timeout

Allcharles committed:
Added Temporary Token

atruskie reviewed (commented):
this token needs to be encrypted in the travis yaml and referenced here as an environment variable

Allcharles committed:
Added Azure Pipelines

atruskie committed:
Merge branch 'master' into linux_build

Allcharles committed:
Potential Babel Fix

Allcharles committed:
Update Babel Version

Allcharles committed:
Force Babel Install

Allcharles committed:
Moved babel install to new script

Allcharles committed:
Incorrect npm install

This could be the reason why babel was failing. `npm run install` does not run the normal install.

Allcharles committed:
npm-pwsh workaround

Because of the issues with windows running pwsh, linux and windows now have seperate ap_download scripts and a node script differentiates between them based on OS.

Allcharles committed:
Debugging AP Downloader

Allcharles committed:
Backdate script_ap to use curl.exe

Allcharles committed:
Added extra error reporting for VM

Allcharles committed:
Potential npm-pwsh workaround

Allcharles committed:
Seperated AP Download from dependencies

Allcharles committed:
Update README.md
